R/opendata_get_resource.R

```diff
@@ -0,0 +1,85 @@
+
+#' Get Open Data resource
+#'
+#' @param res_id The resource ID as found on \href{https://www.opendata.nhs.scot/}{NHS Open Data platform}
+#' @param rows (optional) specify the max number of rows to return
+#' use this when testing code to reduce the size of the request
+#' it will default to all data
+#'
+#' @return a [tibble][tibble::tibble-package] with the data
+#' @export
+#'
+#' @examples opendata_get_resource(res_id = "a794d603-95ab-4309-8c92-b48970478c14")
+opendata_get_resource <- function(res_id, rows = NULL) {
+  if (!opendata_check_res_id(res_id)) {
+    stop(glue::glue("The resource ID supplied ('{res_id}') is invalid"))
+  }
+
+  # Get the max rows allowed by the API if rows isn't specified
+  max_rows <- min(rows, 32000)
```

Comment:
We don`t actually have a 32000 row limit for our CKAN instance, so no need for this line. Instead I think we should default to the total number of rows in the resource if row argument isn`t otherwise specified - that`s the default behaviour I`d expect from a function like this.

Reply:
I think I found that in the API docs here: [https://docs.ckan.org/en/latest/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search](https://docs.ckan.org/en/latest/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search) the default for `limit` is set to 100 (I just tested that) - so if you don't supply an argument i.e. `rows = NULL` it will only return 100 rows.

I'm not sure if it's possible to get the number of rows in a resource without first doing an extra call per resource? Using the 32 000 (which is the default max) just returns all of the rows. It doesn't seem that requesting more rows has any impact on execution time either.

Reply:
Any thoughts on the above? I'm going to mark it ready for review otherwise :)

Reply:
Sorry I think I left a review pending somehow and not sure my comments actually came through? I would go with making an additional one-line query to get the number of rows and use that when rows aren`t specified, to avoid cutting off data from resources larger than 32k rows - see code suggestion above